### PR TITLE
Light app support

### PR DESCRIPTION
**Pull request by raynaudoe**
This PR adds the compiler option to build a "light" Polkadot app version, that only supports the following methods:
```
  'staking.validate',
  'staking.setPayee',
  'staking.chill',
  'staking.nominate',
  'session.setKeys',
  'session.purgeKeys',
  'balances.transfer',
  'balances.transferKeepAlive',
  'staking.bond',
  'staking.bondExtra',
  'staking.unbond',
  'staking.withdrawUnbonded',
  'staking.payoutStakers',
  'staking.rebond',
  'utility.batch',
  'utility.batchAll',
```
To build the light app:
`SUBSTRATE_PARSER_FULL=1 make`

For nano S only:
`make build_full_parser_s`

For nano X only:
`make build_full_parser_x`

### CODE REVIEW

File: app/Makefile [hunk 1]
```diff
@@ -53,6 +53,11 @@ DEFINES += SUPPORT_SR25519
 $(info ************  SR25519 ENABLED ************)
 endif
 
+ifeq ($(SUBSTRATE_PARSER_FULL),1)
+DEFINES += SUBSTRATE_PARSER_FULL
+$(info ************ FULL PARSER ENABLED ************)
+endif
+
 ifndef COIN
```
Comment (jleni) [inline app/Makefile:61]: Let's bump the version number

File: .circleci/config.yml [hunk 2]
```diff
@@ -28,12 +28,12 @@ jobs:
             cd ~/project/app/rust
             cargo clippy --version
             cargo clippy --all-features --all-targets || true
-      - run:
-          name: audit
-          command: |
-            cd ~/project/app/rust
-            cargo audit --version
-            cargo audit
+#      - run:
```
Comment (jleni) [inline .circleci/config.yml:31]: I think we can reenable audit now.. the problems upstream have been fixed! 🎉

**Review by jleni — CHANGES_REQUESTED** [review]

File: app/Makefile (hunk 1 above)
Comment (raynaudoe) [reply app/Makefile:61]: Already bumped to 6.28.1 (up patch version).  is that ok ?

File: .circleci/config.yml (hunk 2 above)
Comment (raynaudoe) [reply .circleci/config.yml:31]: Great! I'll re-enable it

Comment (raynaudoe) [reply .circleci/config.yml]: We still have the same problem with `cargo audit` https://app.circleci.com/pipelines/github/Zondax/ledger-polkadot/311/workflows/71363617-2cac-445d-82e3-ade8a7149a16/jobs/1418/parallel-runs/0/steps/0-104?invite=true

Comment (jleni) [reply .circleci/config.yml]: yes, we need to upgrade dependencies too. 1 min, I commit that

File: app/Makefile (hunk 1 above)
Comment (jleni) [reply app/Makefile:61]: sorry, I didn't notice! That's perfect then

**Pull request merged**